backend/onedrive/onedrive.go

```diff
@@ -374,6 +388,12 @@ func Config(ctx context.Context, name string, m configmap.Mapper, config fs.Conf
 	region, graphURL := getRegionURL(m)
 
 	if config.State == "" {
+		disableSitePermission, ok := m.Get("disable_site_permission")
+		if disableSitePermission == "true" && ok {
```

Comment:
```suggestion
		if ok && disableSitePermission == "true" {
```

Usually we put `ok` before the value predicate. Or we can just remove it here.

Reply:
After some more thinking I think we should just remove the check on `ok` to allow possible changes to the default value. If at some later point we decided `true` will be better as the default, the `ok` check here will break that as it assumes `false` should be the default.